src/lib/logical_query_plan/abstract_lqp_node.cpp

```diff
@@ -135,6 +135,15 @@ void AbstractLQPNode::set_child(LQPChildSide side, const std::shared_ptr<Abstrac
   _child_changed();
 }
 
+uint8_t AbstractLQPNode::child_count() const {
+  /**
+   * Testing the shared_ptrs for null in _children to determine child count
+   */
+  return _children[0] ? (_children[1] ? 2 : 1) : (_children[1] ? 1 : 0);
```

Comment:
Would this make the code clearer?

```c++
return _children.size() - std::count(_children.cbegin(), _children.cend(), nullptr);
```

Reply:
I don't know. If you know `std` well: maybe yes. If you don't: no.

Reply:
i feel like you can look up `std::count` real quick and then its quite clear what it does. Also makes it more robust, as it does not make any assumptions about the contents of `_inputs`

Reply:
Good point.

Reply:
@tbjoern @apopiak So are you going to replace it?